parts/linux/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -384,40 +384,80 @@ should_skip_nvidia_drivers() {
     echo "$should_skip"
 }
 
-start_watch () {
-  capture_time=$(date +%s)
-  start_timestamp=$(date +%H:%M:%S)
+installJq () {
+  # jq is not available until downloaded in install-dependencies.sh with the installDeps function
+  # but it is needed earlier to call the capture_benchmarks function in pre-install-dependencies.sh
+  output=$(jq --version)
+  if [ -n "$output" ]; then
+    echo "$output"
+  else
+    if [[ $OS == $MARINER_OS_NAME ]]; then
+      sudo tdnf install -y jq
+      echo "jq was installed: $(jq --version)"
+    else
+      apt_get_install 5 1 60 jq
+      echo "jq was installed: $(jq --version)"
+    fi
+  fi
 }
 
-stop_watch () {
+capture_benchmarks () {
```

Comment:
This function has 2 purposes, namely, capture benchmarks for a section and for the whole script.
If I were implementing it, I would separate this function into two, e.g., `capture_section_benchmarks` and `capture_script_benchmarks` (or capture_final_section_benchmarks` if it's more appropriate), instead of using a bool to control the logic.
The reasons behind that are
- Single Responsibility Principle: The idea behind the SRP is that every class, module, or function in a program should have one responsibility/purpose in a program (copied from search engine)
- There isn't too much common logic between when $is_final_section == true vs false. So duplicate codes shouldn't be an issue here. 
- Even there were duplicate codes, I would put them into functions so that `capture_section_benchmarks` and `capture_script_benchmarks` can reuse.

Just my 2 cents :)